litestar/dto/factory/abc.py

```diff
@@ -130,24 +135,24 @@ def on_registration(cls, handler_context: HandlerContext) -> None:
             handler_context: A :class:`HandlerContext <.HandlerContext>` instance. Provides information about the
                 handler and application of the DTO.
         """
-        if handler_context.parsed_type.is_subclass_of(DTOData):
-            parsed_type = handler_context.parsed_type.annotation.parsed_type
-        else:
-            parsed_type = handler_context.parsed_type
-
-        if parsed_type.is_collection:
-            if len(parsed_type.inner_types) != 1:
-                raise InvalidAnnotation("AbstractDTOFactory only supports homogeneous collection types")
-            handler_type = parsed_type.inner_types[0]
-        else:
-            handler_type = parsed_type
+        parsed_type = handler_context.parsed_type
+        type_is_pagination_type = is_pagination_type(parsed_type)
```

Comment:
Is there a way to make this more generic? Special casing here doesn't feel like it's ideal. Maybe this can be handled on the pagination type instead or something like that?

Reply:
Yeah, this made it much nicer!